### PR TITLE
checkout updates for ui-extensions

### PR DESCRIPTION
**Pull request by jamesvidler**
### Background
Closes https://github.com/Shopify/checkout-web/issues/22623.

**TLDR**; this brings over the latest changes from the `checkout-web/packages/ui-extensions(-react)` packages. This is identical in features and functionality as version `0.27.0` of the legacy `checkout-ui-extensions` packages which are being published in this [PR](https://github.com/Shopify/ui-extensions/pull/1033).

> **Note**: We do not need to wait for the [legacy package](https://github.com/Shopify/ui-extensions/pull/1033) to publish. The changes in this PR will go to the unstable branch where we can continue to make any other small updates before branching and shipping `2023-04` (according to these [instructions](https://github.com/Shopify/ui-extensions/blob/unstable/documentation/versions-and-deploys.md)).

Docs will not be published for this version.
### Solution
- Ran `yarn prepare-package-release:ui-extensions` which copies over from the private `checkout-web/packages/ui-extensions(-react)` packages.
- Adjusted some lint rules so CI passes.
- Manually [reverted component shared spacing properties](https://github.com/Shopify/ui-extensions/pull/1033/commits/faf139946940ddcf06a0796e3e43bcdd4f4296d8#diff-5254ed708b70eef814c703c490f7b0046ac906b5e66ef8a1b2e059fcd6fd0ee4) like what was needed in the corresponding [legacy package PR](https://github.com/Shopify/ui-extensions/pull/1033)

### 🎩
1. Run `spin up ui-extensions`.
2. Checkout this branch.
3. Run `yarn build` to build the packages.
4. Scaffold a standalone extension (outside of spin) on your machine and adjust the packages to use `"@shopify/ui-extensions-react": "2022.10.6"` and import from `@shopify/ui-extensions-react/checkout`.

Then, run the following command to copy the packages to your local extension to use. This will ask you to choose the spin instance:
```
VM=direct.$(spin show -o fqdn)
rsync -av --delete $VM:~/src/github.com/Shopify/ui-extensions/packages/ui-extensions node_modules/@shopify/ && \
   rsync -av --delete $VM:~/src/github.com/Shopify/ui-extensions/packages/ui-extensions-react node_modules/@shopify/
```

Finally, run `yarn dev` to run your extension in production using this package.

You should not have any type errors relating to the package and your dev extension should render without error.

### Checklist

- [X] I have :tophat:'d these changes
- [ ] I have updated relevant documentation

### CODE REVIEW

**Review by kumar303 — APPROVED** [review]
I tophatted in production with a linked library and could use some newly added APIs 🚀 I wasn't able to review all changes.

**Review by oliverigor — APPROVED** [review]
Mainly checked the code and did some basic tophatting.

File: packages/ui-extensions-react/src/surfaces/checkout/hooks/api.ts [hunk 1]
```diff
@@ -26,9 +31,13 @@ export function useApi<
 }
 
 /**
- * Returns the full API object that was passed in to your
- * extension when it was created.
- * @deprecated you shoud be importing useApi instead
```
Comment (lemonmade) [inline packages/ui-extensions-react/src/surfaces/checkout/hooks/api.ts:31]: Was this removed on purpose?

Comment (jamesvidler) [reply packages/ui-extensions-react/src/surfaces/checkout/hooks/api.ts:31]: Yes, plan is to not mark these as deprecated yet for `2023-04`. `2023-07` will introduce deprecations.
